### PR TITLE
fix: turn off tooltips for touch events

### PR DESCRIPTION
**Pull request by wescopeland**
This PR resolves https://github.com/RetroAchievements/RAWeb/issues/1365. There are various issues with tooltips on mobile because taps are interpreted as hover events. I have confirmed these issues occur not only on iOS, but occur on some Android browsers as well.

A new event listener, `ontouchstart`, is attached to DOM elements responsible for opening tooltips. This listener will always fire before `onmouseover`. If it does fire, a flag is set and tooltips will not open on tap.

This solution has the benefit of not breaking mobile devices that are using trackpads (which is increasingly common with iPads).

### CODE REVIEW

File: public/gameInfo.php [hunk 1]
```diff
@@ -1006,7 +1006,7 @@ function ResetProgress() {
                     echo "</div>";
 
                     echo "<script>var {$containername}tooltip = \"$tooltip\";</script>";
-                    echo "<div style='float: left; clear: left' onmouseover=\"Tip({$containername}tooltip)\" onmouseout=\"UnTip()\">";
+                    echo "<div style='float: left; clear: left' ontouchstart=\"mobileSafeTipEvents.touchStart()\" onmouseover=\"mobileSafeTipEvents.mouseOver({$containername}tooltip)\" onmouseout=\"UnTip()\">";
```
Comment (Jamiras) [inline public/gameInfo.php:1009]: A similar change also needs to be made in activePlayersBootstrap.js

File: public/js/all.js [hunk 2]
```diff
@@ -121,6 +121,31 @@ function replaceAll(find, replace, str) {
   return str.replace(new RegExp(find, 'g'), replace);
 }
 
+let wasTouchDeviceDetected = false;
+function mobileSafeTipTouchStart() {
+  wasTouchDeviceDetected = true;
+}
```
Comment (Jamiras) [inline public/js/all.js:127]: this function appears unused.

Comment (wescopeland) [reply public/js/all.js]: Fixed

**Review by Jamiras — APPROVED** [review]

**Review by luchaos — APPROVED** [review]
Eventually all of the very old wz_tooltip implementation should be replaced with https://popper.js.org/ which would also resolve a lot of the mobile issues (events, positioning)

**Comment by wescopeland** [comment]
@luchaos Floating UI is probably worth looking into as a more modern choice: https://opencollective.com/floating-ui/updates/popper-is-evolving-into-floating-ui

**Pull request merged**